test/unit/utils/UWebSocketTransport.spec.ts

```diff
@@ -10,270 +15,693 @@ describe('UWebSocketTransport', () => {
   let errorListener: jest.Mock
 
   beforeEach(async () => {
-    mockConfig.getNumber.mockImplementation(async (key) => {
-      const defaults: Record<string, number> = {
-        WS_TRANSPORT_MAX_QUEUE_SIZE: 1000,
-        WS_TRANSPORT_QUEUE_DRAIN_TIMEOUT: 5000
-      }
-      return defaults[key] || null
-    })
-
+    // Mock configuration with realistic values
+    mockConfig.getNumber.mockImplementation(
+      async (key) =>
+        ({
+          WS_TRANSPORT_MAX_QUEUE_SIZE: 3,
+          WS_TRANSPORT_RETRY_DELAY_MS: 1000,
+          WS_TRANSPORT_MAX_RETRY_ATTEMPTS: 5,
+          WS_TRANSPORT_MAX_BACKOFF_DELAY_MS: 30000
+        })[key] || null
+    )
+
+    // Initialize mock socket with default behavior
     mockSocket = {
-      send: jest.fn().mockReturnValue(1),
+      send: jest.fn().mockReturnValue(UWebSocketSendResult.SUCCESS),
       close: jest.fn(),
       end: jest.fn(),
       getUserData: jest.fn().mockReturnValue({ isConnected: true })
     } as jest.Mocked<IUWebSocket<{ isConnected: boolean }>>
 
+    // Initialize event emitter and transport
     mockEmitter = mitt<IUWebSocketEventMap>()
-    transport = await createUWebSocketTransport(mockSocket, mockEmitter, mockConfig, mockLogs)
+    transport = await createUWebSocketTransport(mockSocket, mockEmitter, mockConfig, mockLogs, mockMetrics)
     errorListener = jest.fn()
     transport.on('error', errorListener)
+    jest.useFakeTimers({ advanceTimers: true })
   })
 
   afterEach(() => {
     mockEmitter.all.clear()
+    jest.useRealTimers()
+    jest.clearAllMocks() // Ensure mocks are cleared
   })
 
-  describe('Transport Initialization', () => {
-    it('should initialize transport with correct state', () => {
-      expect(transport.isConnected).toBe(true)
+  describe('Transport Lifecycle', () => {
+    it('should initialize with correct state and emit connect event', async () => {
+      // Set up connect listener before creating transport
+      const connectListener = jest.fn()
+      const newEmitter = mitt<IUWebSocketEventMap>()
+
+      // Create new transport instance
+      const newTransport = await createUWebSocketTransport(mockSocket, newEmitter, mockConfig, mockLogs, mockMetrics)
+      newTransport.on('connect', connectListener) // Add listener for completeness, but timing makes it hard to test reliably here
```

Comment:
Isn't the connect emitted before the transport is set? I wouldn't test it if that's the case, we can leave it out.